### PR TITLE
satellite/payments: Improve Invoice Generation 

### PR DESCRIPTION
**Pull request by stefanbenten**
What: 
- Adds a description on the invoice
- Adds separate Invoice Item Lines for Storage, Bandwidth Usage and Object Fee
- Improves the Billing History Listing
- It also adds a config flag that disables the Automatic Processing that stripe offers per default.

Why: It was confusing and misleading.

Please describe the performance impact: few minor remote requests to the stripe API.

## Code Review Checklist (to be filled out by reviewer)
 - [x] NEW: Are there any Satellite database migrations? Are they forwards _and_ backwards compatible? 
 - [x] Does the PR describe what changes are being made?
 - [x] Does the PR describe why the changes are being made?
 - [x] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [x] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [x] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [x] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [x] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [x] Does any documentation need updating?
 - [x] Do the database access patterns make sense?

### CODE REVIEW

File: satellite/payments/stripecoinpayments/invoices.go [hunk 1]
```diff
@@ -43,7 +43,6 @@ func (invoices *invoices) List(ctx context.Context, userID uuid.UUID) (invoicesL
 			Amount:      stripeInvoice.AmountDue,
 			Status:      string(stripeInvoice.Status),
 			Link:        stripeInvoice.InvoicePDF,
-			End:         time.Unix(stripeInvoice.PeriodEnd, 0),
```
Comment (stefanbenten) [inline satellite/payments/stripecoinpayments/invoices.go]: At least from what Jens and me found, this only shows the time at which the generation process of the invoice on stripe side finished.

File: satellite/payments/stripecoinpayments/service.go [hunk 2]
```diff
@@ -583,20 +589,41 @@ func (service *Service) createInvoiceItems(ctx context.Context, cusID, projName
 
 	projectPrice := service.calculateProjectUsagePrice(record.Egress, record.Storage, record.Objects)
 
-	projectItem := &stripe.InvoiceItemParams{
-		Amount:      stripe.Int64(projectPrice.TotalInt64()),
-		Currency:    stripe.String(string(stripe.CurrencyUSD)),
-		Customer:    stripe.String(cusID),
-		Description: stripe.String(fmt.Sprintf("project %s", projName)),
+	projectItemBase := &stripe.InvoiceItemParams{
```
Comment (rikysya) [inline satellite/payments/stripecoinpayments/service.go]: I think you should change this variable to be value type, or later introduce of new item variables doesn't make sense as you change the same reference all the time

File: satellite/payments/stripecoinpayments/service.go [hunk 3]
```diff
@@ -837,10 +864,22 @@ func (service *Service) CreateInvoices(ctx context.Context) (err error) {
 func (service *Service) createInvoice(ctx context.Context, cusID string) (err error) {
 	defer mon.Task()(&ctx)(&err)
 
+	// get the first invoice item's period
+	iter := service.stripeClient.InvoiceItems.List(&stripe.InvoiceItemListParams{Customer: stripe.String(cusID)})
+	if !iter.Next() {
+		return Error.New("no invoice items found, no invoice to be created")
```
Comment (rikysya) [inline satellite/payments/stripecoinpayments/service.go]: If you won't to throw this error you might also want to remove error handling on line 887

File: satellite/payments/stripecoinpayments/service.go [hunk 4]
```diff
@@ -837,10 +859,26 @@ func (service *Service) CreateInvoices(ctx context.Context) (err error) {
 func (service *Service) createInvoice(ctx context.Context, cusID string) (err error) {
 	defer mon.Task()(&ctx)(&err)
 
+	var description string
+	// get the first invoice item's period
+	iter := service.stripeClient.InvoiceItems.List(&stripe.InvoiceItemListParams{Customer: stripe.String(cusID)})
+	for iter.Next() {
+		if iter.Err() != nil {
```
Comment (rikysya) [inline satellite/payments/stripecoinpayments/service.go]: iter.Err should be called after `iter.Next`

File: satellite/payments/stripecoinpayments/service.go [hunk 5]
```diff
@@ -837,10 +856,26 @@ func (service *Service) CreateInvoices(ctx context.Context) (err error) {
 func (service *Service) createInvoice(ctx context.Context, cusID string) (err error) {
 	defer mon.Task()(&ctx)(&err)
 
+	var description string
+	// get the first invoice item's period
+	iter := service.stripeClient.InvoiceItems.List(&stripe.InvoiceItemListParams{Customer: stripe.String(cusID)})
```
Comment (rikysya) [inline satellite/payments/stripecoinpayments/service.go]: you can set limit to fetch only one entry

File: satellite/payments/stripecoinpayments/service.go [hunk 6]
```diff
@@ -583,20 +589,41 @@ func (service *Service) createInvoiceItems(ctx context.Context, cusID, projName
...
 		Period: &stripe.InvoiceItemPeriodParams{
-			End:   stripe.Int64(record.PeriodEnd.Unix()),
 			Start: stripe.Int64(record.PeriodStart.Unix()),
+			End:   stripe.Int64(record.PeriodEnd.Unix()),
 		},
 	}
+	projectItemBase.AddMetadata("projectID", record.ProjectID.String())
 
-	projectItem.AddMetadata("projectID", record.ProjectID.String())
+	projectStorage := projectItemBase
```
Comment (rikysya) [inline satellite/payments/stripecoinpayments/service.go]: this doesn't make sense as it is a pointer to the same value

**Pull request merged**